File: reservation-service/domain/priceVariation.go
package domain

import (
	"time"

	"github.com/gocql/gocql"
)

type PriceVariation struct {
	VariationID gocql.UUID `json:"variationId"`
	Location    string     `json:"location"`
	AccommID    string     `json:"accommId"`
	StartDate   time.Time  `json:"startDate"`
	EndDate     time.Time  `json:"endDate"`
	Percentage  int        `json:"percentage"`
}

File: front-end-service/src/app/model/priceVariation.ts
import { formatDate } from "@angular/common";

export class PriceVariation{
    variationId: string;
    location: string;
    accommId: string;
    startDate: Date;
    endDate: Date;
    percentage: number;
    constructor() {
        this.variationId = ""
        this.location = ""
        this.accommId = ""
        this.percentage = 0
        this.startDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
        this.endDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
    }
}